server/routes/timelineResolver.ts

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable no-param-reassign */
+import type { TimelineResponse, TimelineEventResponse } from 'educationAndWorkPlanApiClient'
+import type { TimelineEvent } from 'viewModels'
+import dateComparator from './dateComparator'
+
+const filterTimelineEvents = (timelineResponse: TimelineResponse): TimelineResponse => {
```

Comment:
Your function arg here should not be the API Response type, it needs to be the view model type (`TimeLine`) - it's what comes out of the service method that is being passed into this function 👍

Reply:
And therefore your test above needs to change to use Timeline rather than TimelineResponse, at which point you'll have a `timestamp` field that is a `Date` which should then sort as you expect it to